File: server/dotnet/sdk-dotnet-example/Infrastructure/Mappers/PaymentDetails/MobilePaymentMethodSpecificOutputMapper.cs
using  MobilePaymentMethodSpecificOutputDto = Business.Domain.Payments.PaymentDetails.MobilePaymentMethodSpecificOutput;
using  MobilePaymentMethodSpecificOutputSdk = OnlinePayments.Sdk.Domain. MobilePaymentMethodSpecificOutput;

namespace Infrastructure.Mappers.PaymentDetails;

public static class MobilePaymentMethodSpecificOutputMapper
{
    public static MobilePaymentMethodSpecificOutputDto Map( MobilePaymentMethodSpecificOutputSdk? response)
    {
        return new()
        {
            Network = response?.Network,
            AuthorisationCode = response?.AuthorisationCode,
            PaymentProductId = response?.PaymentProductId,
            FraudResults = CardFraudResultsMapper.Map(response?.FraudResults),
            PaymentData = MobilePaymentDataMapper.Map(response?.PaymentData),
            ThreeDSecureResults = ThreeDSecureResultsMapper.Map(response?.ThreeDSecureResults)
        };
    }
}

File: server/php/src/Infrastructure/Mappers/PaymentDetails/MobilePaymentMethodSpecificOutputMapper.php
<?php

namespace OnlinePayments\ExampleApp\Infrastructure\Mappers\PaymentDetails;

use OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails\MobilePaymentMethodSpecificOutput as MobilePaymentMethodSpecificOutputDto;
use OnlinePayments\Sdk\Domain\MobilePaymentMethodSpecificOutput as MobilePaymentMethodSpecificOutputSdk;

final class MobilePaymentMethodSpecificOutputMapper
{
    public static function mapFromSdkResponse(?MobilePaymentMethodSpecificOutputSdk $response): ?MobilePaymentMethodSpecificOutputDto
    {
        if ($response === null) {
            return null;
        }

        $dto = new MobilePaymentMethodSpecificOutputDto();
        $dto->network = $response?->network ?? null;
        $dto->authorisationCode = $response?->authorisationCode ?? null;
        $dto->paymentProductId = $response?->paymentProductId ?? null;
        $dto->fraudResults = CardFraudResultsMapper::mapFromSdkResponse($response?->fraudResults);
        $dto->paymentData = MobilePaymentDataMapper::mapFromSdkResponse($response?->paymentData);
        $dto->threeDSecureResults = ThreeDSecureResultsMapper::mapFromSdkResponse($response?->threeDSecureResults);

        return $dto;
    }
}
